RecoBTag/SecondaryVertex/plugins/BoostedDoubleSVProducer.cc

```diff
@@ -0,0 +1,773 @@
+                      tau2_trackSip3dSig_1 = IP3Ds_2.at(1);
+
+     }
+
+     math::XYZVector jetDir = jet->momentum().Unit();
+     reco::TrackKinematics tau1Kinematics;
+     reco::TrackKinematics tau2Kinematics;
+     std::vector<float> tau1_trackEtaRels, tau2_trackEtaRels;
+
+     std::map<double, size_t> VTXmap;
+     for (size_t vtx = 0; vtx < svTagInfo.nVertices(); ++vtx)
+     {
+       const reco::VertexCompositePtrCandidate vertex = svTagInfo.secondaryVertex(vtx);
+       // get the vertex kinematics
+       reco::TrackKinematics vertexKinematic(vertex);
+
+       if (currentAxes.size() > 1)
+       {
+               if (reco::deltaR2(svTagInfo.flightDirection(vtx),currentAxes[1]) < reco::deltaR2(svTagInfo.flightDirection(vtx),currentAxes[0]))
+               {
+                       tau2Kinematics = tau2Kinematics + vertexKinematic;
+                       if( tau2_flightDistance2dSig < 0 )
+                       {
+                         tau2_flightDistance2dSig = svTagInfo.flightDistance(vtx,true).significance();
+                         tau2_vertexDeltaR = reco::deltaR(svTagInfo.flightDirection(vtx),currentAxes[1]);
+                       }
+                       etaRelToTauAxis(vertex, currentAxes[1], tau2_trackEtaRels);
+                       tau2_nSecondaryVertices += 1.;
+               }
+               else
+               {
+                       tau1Kinematics = tau1Kinematics + vertexKinematic;
+                       if( tau1_flightDistance2dSig < 0 )
+                       {
+                         tau1_flightDistance2dSig =svTagInfo.flightDistance(vtx,true).significance();
+                         tau1_vertexDeltaR = reco::deltaR(svTagInfo.flightDirection(vtx),currentAxes[0]);
+                       }
+                       etaRelToTauAxis(vertex, currentAxes[0], tau1_trackEtaRels);
+                       tau1_nSecondaryVertices += 1.;
+               }
+
+       }
+       else if (currentAxes.size() > 0)
+       {
+               tau1Kinematics = tau1Kinematics + vertexKinematic;
+               if( tau1_flightDistance2dSig < 0 )
+               {
+                 tau1_flightDistance2dSig =svTagInfo.flightDistance(vtx,true).significance();
+                 tau1_vertexDeltaR = reco::deltaR(svTagInfo.flightDirection(vtx),currentAxes[0]);
+               }
+               etaRelToTauAxis(vertex, currentAxes[0], tau1_trackEtaRels);
+               tau1_nSecondaryVertices += 1.;
+       }
+
+       GlobalVector flightDir = svTagInfo.flightDirection(vtx);
+       if (reco::deltaR2(flightDir, jetDir)<(maxSVDeltaRToJet_*maxSVDeltaRToJet_))
+         VTXmap[svTagInfo.flightDistance(vtx).error()]=vtx;
+     }
+     nSV = VTXmap.size();
+
+
+     math::XYZTLorentzVector allSum = allKinematics.weightedVectorSum() ;
+     if ( tau1_nSecondaryVertices > 0. )
+     {
+       math::XYZTLorentzVector tau1_vertexSum = tau1Kinematics.weightedVectorSum();
+       tau1_vertexEnergyRatio = tau1_vertexSum.E() / allSum.E();
```

Comment:
Are there pathological cases where `allKinematics` is empty, so `allSum.E() == 0`? It looks like `trackSize` being 0 could cause this situation.